### PR TITLE
Improve onboarding experience

### PR DESCRIPTION
**Pull request by lorenzoh**
This adds a contributor guide (leaning on Flux.jl's), makes some updates to the README, and adds a developer guide for running tests and documentation.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# Contributor guide for FastAI.jl
+
+First off, thank you for considering to contribute to FastAI.jl. We welcome contributions and are happy to work with you!
```
Comment (darsnack) [inline CONTRIBUTING.md:3]:
```suggestion
First off, thank you for considering contributing to FastAI.jl. We welcome contributions and are happy to work with you!
```

File: DEVELOPING.md [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+- Jupyter Notebooks should be used when they use resources that are not available on the GitHub CI, like a GPU needed for training. You should run them locally and the outputs will be captured and inserted into the HTML page.
+- Markdown documents should be preferred for everything else, as they allow the code examples to be run on the GitHub CI, meaning they'll stay up-to-date unlike a notebook that has to be manually rerun.
+
+Both formats support the [Markdown syntax of Publish.jl](https://michaelhatherly.github.io/Publish.jl/dev/docs/syntax.html) and in markdown files the [cell syntax of Publish.jl](https://michaelhatherly.github.io/Publish.jl/dev/docs/cells.html) can be used to mark code cells. These will be run and the output is inserted into the HTML page.
+
+### Linking to documentation 
+
+For a new documentation file to be discoverable, you have to add an entry to the nested Markdown list in `toc.md`, which corresponds to the sidebar in the documentation (Updating the sidebar currently requires interrupting and reincluding the file that starts the development server).
```
Comment (darsnack) [inline DEVELOPING.md:87]:
```suggestion
For a new documentation file to be discoverable, you have to add an entry to the nested Markdown list in `toc.md`, which corresponds to the sidebar in the documentation _(updating the sidebar currently requires interrupting and reincluding the file that starts the development server)_.
```

File: README.md [hunk 3]
```diff
@@ -1,27 +1,59 @@
...
+To dive in, you may be interested in
+
+- an [overview of the high-level API](https://fluxml.ai/FastAI.jl/dev/i/?id=documents%2Fdocs%2Fintroduction.md),
+- seeing some [example learning tasks](https://fluxml.ai/FastAI.jl/dev/i/?id=documents%2Fnotebooks%2Fquickstart.ipynb),
+- finding out [how you can search for and find datasets and other functionality](https://fluxml.ai/FastAI.jl/dev/i/?id=documents%2Fdocs%2Fdiscovery.md); or
+- [our contributor guide](CONTRIBUTING.md)
+
+## Get in touch
+
+You can get in touch here on GitHub or on the JuliaLang Zulip in the [`#ml-ecosystem` channel](https://julialang.zulipchat.com/#narrow/stream/237432-ml-contributers).
```
Comment (darsnack) [inline README.md:52]:
```suggestion
You can get in touch here on GitHub or on the JuliaLang Zulip in the [`#ml-contributors` channel](https://julialang.zulipchat.com/#narrow/stream/237432-ml-contributers).
```

**Review by darsnack — COMMENTED** [review]
What's the difference between `DEVELOPING.md` and `developing.md`?

**Comment by lorenzoh** [comment]
Thanks for the comments! 

> What's the difference between DEVELOPING.md and developing.md?

That was just an old version that wasn't renamed; i removed it

**Pull request merged**